### DISCUSSION TITLE
SPIKE Prime light intensity values LEGO App - Pybricks

### DISCUSSION THREAD

**Discussion by Aldeatron** [pybricks/support#1172] [General]
Hi!

We´ve noticed with SPIKE Prime that, for the same distance sensor - mat (~16 mm), white gives light intensity almost 100 with LEGO App, and almost 50 with Pybricks. We think it could be because Pybricks measures intensity with sensor brightness at 50% by default. Could we change that % while measuring light intensity?

Thank you!

**Comment by laurensvalk** [comment]
> mat (~16 mm), white gives light intensity almost 100 with LEGO App, and almost 50 with Pybricks. 

Does that mean you can't get any information if the sensor is closer than 16 mm in the LEGO app?

It would seem beneficial if you could measure things even up close.

> Could we change that % while measuring light intensity?

What would you like to change? You could scale the value, if you like. How about doing `* 2`?

**Reply by Aldeatron** [reply]
Thanks for your quick answer!

> Does that mean you can't get any information if the sensor is closer than 16 mm in the LEGO app?

No, it doesn´t. We meant that for the recommended gap by LEGO of 16 mm we have those readings. We will test sensors closer to the mat and see how it goes. :)

**Reply by laurensvalk** [reply]
You don't have to put it closer to the mat, though. If you were using 16mm before, you can keep it that way with Pybricks.

We may scale it slightly differently so we don't cut off any data, but ultimately you're getting the same sensor accuracy.

But now you have the _option_ to put it closer, which may give you some more build options too.

**Reply by Aldeatron** [reply]
Thank you for your help!

We´ve managed to calibrate each light sensor, assigning 100 to White and 0 to Black, so each reading is scaled by a factor (different for each sensor), whatever the gap is. So, at the beginning of each match, a calibration must be carried out for each sensor, as light conditions can be different.

Each factor is usually between 2 and 2.5.

We only need to find out how to store those values as we asked in #1167 ;)

One way to avoid storing values could be coding everything in one main menu program.
